daphne/src/roles_test.rs

```diff
@@ -512,34 +564,117 @@ async fn http_post_aggregate_init_expired_task(version: DapVersion) {
 
     let report = t.gen_test_report(&t.expired_task_id).await;
     let report_share = ReportShare {
-        metadata: report.metadata,
+        report_metadata: report.report_metadata,
         public_share: report.public_share,
         encrypted_input_share: report.encrypted_input_shares[1].clone(),
     };
     let req = t
-        .gen_test_agg_init_req(&t.expired_task_id, vec![report_share])
+        .gen_test_agg_job_init_req(&t.expired_task_id, version, vec![report_share])
         .await;
 
     let resp = t.helper.http_post_aggregate(&req).await.unwrap();
-    let agg_resp = AggregateResp::get_decoded(&resp.payload).unwrap();
-    assert_eq!(agg_resp.transitions.len(), 1);
+    let agg_job_resp = AggregationJobResp::get_decoded(&resp.payload).unwrap();
+    assert_eq!(agg_job_resp.transitions.len(), 1);
     assert_matches!(
-        agg_resp.transitions[0].var,
+        agg_job_resp.transitions[0].var,
         TransitionVar::Failed(TransitionFailure::TaskExpired)
     );
 }
 
 async_test_versions! { http_post_aggregate_init_expired_task }
 
+// Test that the Helper rejects reports with a bad round id
+async fn http_post_aggregate_bad_round(version: DapVersion) {
+    let t = Test::new(version);
+    if version == DapVersion::Draft02 {
+        // Nothing to test.
+        return;
+    }
+
+    let report = t.gen_test_report(&t.time_interval_task_id).await;
+    let report_share = ReportShare {
+        report_metadata: report.report_metadata,
+        public_share: report.public_share,
+        encrypted_input_share: report.encrypted_input_shares[1].clone(),
+    };
+    let req = t
+        .gen_test_agg_job_init_req(&t.time_interval_task_id, version, vec![report_share])
+        .await;
+    let agg_job_id = match &req.resource {
+        DapResource::AggregationJob(agg_job_id) => agg_job_id.clone(),
+        _ => panic!("agg_job_id resource missing!"),
+    };
+    let resp = t.helper.http_post_aggregate(&req).await.unwrap();
+    let agg_job_resp = AggregationJobResp::get_decoded(&resp.payload).unwrap();
+    assert_eq!(agg_job_resp.transitions.len(), 1);
+    assert_matches!(agg_job_resp.transitions[0].var, TransitionVar::Continued(_));
+    // Test wrong round
+    let req = t
+        .gen_test_agg_job_cont_req_with_round(
+            &MetaAggregationJobId::Draft04(Cow::Borrowed(&agg_job_id)),
+            Vec::default(),
+            Some(2),
+        )
+        .await;
+    assert_matches!(
+        t.helper.http_post_aggregate(&req).await,
+        Err(DapAbort::RoundMismatch)
+    );
+}
+
+async_test_versions! { http_post_aggregate_bad_round }
+
+// Test that the Helper rejects reports with a bad round id
```

Comment:
Is this test needed? The only difference with the last test is that here the round indicated by the Leader is zero.

Reply:
I tested both because the error the standard requires for 0 is different than for other values.